src/bson.rs

```diff
@@ -698,6 +698,20 @@ impl Bson {
                 }
             }
 
+            ["$numberDecimalBytes"] => {
```

Comment:
Unintuitively, this function is used in the existing serializer to convert between the intermediate serialized form (i.e. and "extended document") and the final serialized form. Now that decimal128 can serialize this way, this needed to be updated to interpret it.